beat/covariance.py

```diff
@@ -305,6 +305,48 @@ def get_data_covariances(
 
         return cov_ds
 
+    def get_spectradata_covariances(
```

Comment:
this can also likely be combined with the get_data_covariances, you already have the flag freq_domain included ;)